{"version":3,"sources":["../../test/validation_test.js"],"names":["assert","require","User","describe","it","user","name","undefined","result","validateSync","message","errors","done","save","catch","fromSave"],"mappings":";;AAAA;;;AAGA,IAAMA,SAASC,QAAQ,QAAR,CAAf;AACA,IAAMC,OAAOD,QAAQ,aAAR,CAAb;;AAEAE,SAAS,oBAAT,EAA+B,YAAM;AAClCC,OAAG,sBAAH,EAA2B,YAAM;AAC7B,YAAMC,OAAO,IAAIH,IAAJ,CAAS,EAACI,MAAMC,SAAP,EAAT,CAAb;AACA;;AAEA,YAAMC,SAASH,KAAKI,YAAL,EAAf;;AAEA;AACA;AACA;AACA,YAAMC,UAAUF,OAAOG,MAAP,CAAcL,IAAd,CAAmBI,OAAnC;;AAEAV,eAAOU,YAAY,qBAAnB;AACH,KAZD;;AAcAN,OAAG,+CAAH,EAAoD,YAAM;AACrD,YAAMC,OAAO,IAAIH,IAAJ,CAAS,EAACI,MAAM,IAAP,EAAT,CAAb;AACA,YAAME,SAASH,KAAKI,YAAL,EAAf;AACA,YAAMC,UAAUF,OAAOG,MAAP,CAAcL,IAAd,CAAmBI,OAAnC;;AAEAV,eAAOU,YAAY,wCAAnB;AACJ,KAND;;AAQAN,OAAG,4CAAH,EAAiD,UAACQ,IAAD,EAAU;AACtD,YAAMP,OAAO,IAAIH,IAAJ,CAAS,EAACI,MAAM,IAAP,EAAT,CAAb;AACAD,aAAKQ,IAAL,GACKC,KADL,CACW,UAACC,QAAD,EAAc;AACjB,gBAAML,UAAUK,SAASJ,MAAT,CAAgBL,IAAhB,CAAqBI,OAArC;AACAV,mBAAOU,YAAY,wCAAnB;AACAE;AACH,SALL;AAMJ,KARD;AASF,CAhCD","file":"validation_test.js","sourcesContent":["/**\n * Created by david on 4/22/17.\n */\nconst assert = require('assert');\nconst User = require('../src/user');\n\ndescribe('Validating records', () => {\n   it('requires a user name', () => {\n       const user = new User({name: undefined});\n       // validate is a synchronous process; its happening instantly we don't to wait.  If it took a while we'd need asynchronous validation and would use validate()\n\n       const result = user.validateSync();\n\n       // **NOTE** this is the route to the error message that you made in the schema\n       // es6 equivalent 'const { message } = result.errors.name;'\n       // because the const name is the same as the property name\n       const message = result.errors.name.message;\n\n       assert(message === 'User name required.');\n   });\n\n   it('requires a user name longer than 2 characters', () => {\n        const user = new User({name: 'Al'});\n        const result = user.validateSync();\n        const message = result.errors.name.message;\n\n        assert(message === \"Name must be longer than 2 characters.\");\n   });\n\n   it('disallows invalid records from being saved', (done) => {\n        const user = new User({name: 'Al'});\n        user.save()\n            .catch((fromSave) => {\n                const message = fromSave.errors.name.message;\n                assert(message === 'Name must be longer than 2 characters.');\n                done();\n            });\n   });\n});"]}